lib-multisrc/comicaso/src/eu/kanade/tachiyomi/multisrc/comicaso/Comicaso.kt

```diff
@@ -0,0 +1,502 @@
+package eu.kanade.tachiyomi.multisrc.comicaso
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.json.JSONArray
+import org.json.JSONObject
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+abstract class Comicaso(
+    override val name: String,
+    override val baseUrl: String,
+    override val lang: String,
+    private val pageSize: Int = 16,
+    private val excludedGenres: Set<String> = emptySet(),
+) : ParsedHttpSource() {
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+
+    private val dateFormat = SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH)
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Referer", "$baseUrl/")
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = baseUrl.toHttpUrl().newBuilder()
+            .addPathSegment("wp-json")
+            .addPathSegment("neoglass")
+            .addPathSegment("v1")
+            .addPathSegment("mangas")
+            .addQueryParameter("paged", page.toString())
+            .addQueryParameter("per_page", pageSize.toString())
+            .build()
+
+        return GET(url, headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val json = JSONObject(response.body.string())
+        val items = json.getJSONArray("items")
+        return MangasPage(parseMangaList(items), items.length() == pageSize)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/v2/?page=$page"
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(latestUpdatesSelector()).map { element ->
+            latestUpdatesFromElement(element)
+        }
+        return MangasPage(mangas, document.select(latestUpdatesNextPageSelector()).isNotEmpty())
+    }
+
+    override fun latestUpdatesSelector() = "div.ng-list div.ng-list-item"
+
+    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
+        val thumb = element.selectFirst("div.ng-list-thumb")
+        val link = thumb?.selectFirst("a")?.attr("href") ?: ""
+        title = element.selectFirst("div.ng-list-info a h3")!!.text().trim()
```

Comment:
`.text` is already trimmed.
```suggestion
        title = element.selectFirst("div.ng-list-info a h3")!!.text()
```

Suggestion applies throughout this file.